milestones/api.py

```diff
@@ -163,7 +163,10 @@ def get_course_milestones(course_key, relationship=None):
     _validate_course_key(course_key)
 
     if relationship is not None:
-        _validate_milestone_relationship_type(relationship)
+        try:
+            _validate_milestone_relationship_type(relationship)
+        except exceptions.InvalidMilestoneRelationshipTypeException:
+            return []
```

Comment:
If the model isn't providing a benefit to us, then I'm all for removing it -- we originally added the model because we anticipated more use cases around relationship types, but it seems that has not materialized to-date.